lib/resources/nginx_conf.rb

```diff
@@ -93,8 +96,9 @@ def resolve_references(data, rel_path)
       if data.key?('include')
         data.delete('include').flatten
             .map { |x| File.expand_path(x, rel_path) }
+            .map { |x| find_files(x) }.flatten
             .map { |path| parse_nginx(path) }
-            .map { |e| data.merge!(e) }
+            .map { |e| data.merge!(e) { |_, v1, v2| v1 + v2 } }
```

Comment:
I feel like this is wrong...just not sure how to make it better. When loading server configs from other files (e.g. `/etc/nginx/conf.d/*.conf`) the `merge!` would clobber the `server` value and replace it. This appends the list to the key...but that assumes that `v1` and `v2` will always be lists...I don't like that assumption.

Reply:
should probably be an `each`; also let's extract this piece as it feels to cover some complexity (for deep merging)

Reply:
By "lists" to you mean `Array`s?  Why wouldn't they be? If that's a concern, how do we protect against it?

We own the entire codepath here, so if there's additional promises that need to be made, we can make them.